python/dolfinx/wrappers/caster_petsc.h

```diff
@@ -1,30 +1,37 @@
-// Copyright (C) 2017-2021 Chris Richardson and Garth N. Wells
+// Copyright (C) 2017-2023 Chris Richardson and Garth N. Wells
 //
 // This file is part of DOLFINx (https://www.fenicsproject.org)
 //
 // SPDX-License-Identifier:    LGPL-3.0-or-later
 
 #pragma once
 
+#include <nanobind/nanobind.h>
 #include <petsc4py/petsc4py.h>
 #include <petscdm.h>
 #include <petscis.h>
 #include <petscksp.h>
 #include <petscmat.h>
 #include <petscsnes.h>
 #include <petscvec.h>
-#include <pybind11/pybind11.h>
 
-// pybind11 casters for PETSc/petsc4py objects
+// nanobind casters for PETSc/petsc4py objects
 
-namespace py = pybind11;
+namespace nb = nanobind;
 
 // Import petsc4py on demand
-#define VERIFY_PETSC4PY(func)                                                  \
+#define VERIFY_PETSC4PY0(func)                                                 \
   if (!func)                                                                   \
   {                                                                            \
     if (import_petsc4py() != 0)                                                \
-      throw std::runtime_error("Error when importing petsc4py");               \
+      return false;                                                            \
+  }
+
+#define VERIFY_PETSC4PY1(func)                                                 \
```

Comment:
`VERIFY_PETSC4PY1` -> `VERIFY_PETSC4PY_HANDLE_RETURN`? At first glance, the difference between the two was not evident to me.